### DISCUSSION TITLE
Render NBSP and NNBSP chars

### DISCUSSION THREAD

**Discussion by Fred-Vatin** [monkeytypegame/monkeytype#4793] [Ideas]
In French, the typography rules required using NBSP or more often [NNBSP](https://en.wikipedia.org/wiki/Thin_space) (narrow unbreakable space) before or after some punctuation : `? ! « » ; :` or to separate numbers like in `1 000 000`. The online http://dactylotest.free.fr/dactylotest offers the option to display those chars.
![image](https://github.com/monkeytypegame/monkeytype/assets/22337329/efea24c3-5a9d-406e-b374-cc596a3cd80e)

Monkeytype could do the same with a little box with very thin borders which colors would differ between NBSP and NNBSP.

**Comment by Miodec** [comment]
Im pretty sure spaces are already added in quotes and when enabling punctuation mode.

**Comment by Fred-Vatin** [comment]
I enable puncutation mode with this custom text, `remove fancy typo` = OFF.
```
Here is a NNBSP : before the colon <-
Here is a NBSP : before the colon <-
```
![image](https://github.com/monkeytypegame/monkeytype/assets/22337329/f5137087-8ead-41c9-a56d-522c89c79a4b)

They are not different from the normal space.
The problem is even with `remove fancy typo` = OFF, NBSP and NNBSP are removed when clicking OK.

When I get back to the change custom dialog, my text is transformed as 
```
Here is a NNBSP : before the colon <-
 Here is a NBSP : before the colon <-
```

NBSP and NNBSP were removed. And why a space is added before any new line ? Is it a bug or feature ? `Replace new lines with spaces` is OFF.

**Reply by Miodec** [reply]
The space after newline is not intended i think, but it wont cause issues because its removed on test generation. 

The different types of spaces are removed to avoid confusion. Beucase of the way the words are displayed, spaces between words don't use the actual characters - meaning, there is no distinction between the different types of space characters.

**Reply by Fred-Vatin** [reply]
Hence my feature request to show them with custom mode because they are important in french typography. And enforce the user to type the right space kind.
